runner/src/integrationTest/kotlin/com/cosmotech/runner/service/RunnerServiceRBACTest.kt

```diff
@@ -2820,32 +2821,90 @@ class RunnerServiceRBACTest : CsmRedisTestBase() {
                       role = ROLE_ADMIN)
               val runnerSaved =
                   runnerApiService.createRunner(organizationSaved.id!!, workspaceSaved.id!!, runner)
-              every { getCurrentAccountIdentifier(any()) } returns TEST_USER_MAIL
 
-              if (shouldThrow) {
-                val exception =
-                    assertThrows<CsmAccessForbiddenException> {
-                      runnerApiService.addRunnerAccessControl(
-                          organizationSaved.id!!,
-                          workspaceSaved.id!!,
-                          runnerSaved.id!!,
-                          RunnerAccessControl("id", ROLE_ADMIN))
-                    }
+              assertDoesNotThrow {
                 assertEquals(
-                    "RBAC ${datasetSaved.id!!} - User does not have permission $PERMISSION_WRITE_SECURITY",
-                    exception.message)
-              } else {
-                assertDoesNotThrow {
-                  runnerApiService.addRunnerAccessControl(
-                      organizationSaved.id!!,
-                      workspaceSaved.id!!,
-                      runnerSaved.id!!,
-                      RunnerAccessControl("id", ROLE_ADMIN))
-                }
+                    true,
+                    datasetSaved.security
+                        ?.accessControlList
+                        ?.filter { datasetAccessControl ->
+                          datasetAccessControl.id == TEST_USER_MAIL
+                        }
+                        ?.any { datasetAccessControl -> datasetAccessControl.role == role })
+
+                runnerApiService.addRunnerAccessControl(
+                    organizationSaved.id!!,
+                    workspaceSaved.id!!,
+                    runnerSaved.id!!,
+                    RunnerAccessControl(TEST_USER_MAIL, ROLE_ADMIN))
+
+                val datasetWithUpgradedACL =
+                    datasetApiService.findDatasetById(organizationSaved.id!!, datasetSaved.id!!)
+                assertEquals(
+                    true,
+                    datasetWithUpgradedACL.security
+                        ?.accessControlList
+                        ?.filter { datasetAccessControl ->
+                          datasetAccessControl.id == TEST_USER_MAIL
+                        }
+                        ?.any { datasetAccessControl -> datasetAccessControl.role == role })
               }
             }
           }
 
+  @Test
+  fun `test Dataset RBAC addRunnerAccessControl with new ACL entry`() {
```

Comment:
Could use a comment. this test that a user with a role added on a runner is also add on the dataset. that's a complex behavior.